src/prognostic_equations/edmfx_tke.jl

```diff
@@ -0,0 +1,62 @@
+#####
+##### EDMF SGS flux
+#####
+
+edmfx_tke_tendency!(Yₜ, Y, p, t, colidx, turbconv_model) = nothing
+
+function edmfx_tke_tendency!(
+    Yₜ,
+    Y,
+    p,
+    t,
+    colidx,
+    turbconv_model::DiagnosticEDMFX,
+)
+
+    FT = Spaces.undertype(axes(Y.c))
+    n = n_mass_flux_subdomains(turbconv_model)
+    (; params) = p
+    turbconv_params = CAP.turbconv_params(params)
+    c_d = TCP.tke_diss_coeff(turbconv_params)
+    c_m = TCP.tke_ed_coeff(turbconv_params)
+    vkc = TCP.von_karman_const(turbconv_params)
+    (; ᶜentr_detrʲs, ᶠu³ʲs) = p
+    (; ᶠu³⁰, ᶜshear², ᶜlinear_buoygrad, ᶜtke⁰, ᶜmixing_length, ᶜK_u, ᶜK_h) = p
+    (; ustar) = p.sfc_conditions
+    ᶠgradᵥ = Operators.GradientC2F()
+
+    ᶠρaK_u = p.ᶠtemp_scalar
+    ρatke_flux = p.sfc_temp_C3
+    if true #p.atmos.prognostic_tke
+        # turbulent transport
+        @. ᶠρaK_u[colidx] = ᶠinterp(Y.c.ρ[colidx]) * ᶠinterp(ᶜK_u[colidx])
+        @. ρatke_flux[colidx] = C3(FT(0))
```

Comment:
Need to add surface fluxes. It's a bit annoying as 
```
ρatke_flux = WVector(Fields.level(Y.c.ρ, 1) * constant * ustar^2 * norm(Fields.level(Y.c.uₕ), 1))
```
(or convert it to Covariant3Vector, which will be more correct) But the variables are not on the same spaces (some are on centers and some are on faces), so we may need to use the `DataLayouts`? @charleskawczynski @dennisYatunin any idea?

Reply:
This, it seems, is a generic problem with surface fluxes that should have a generic solution. 

Also, the horizontal velocity norm should be that of the velocity parallel to the surface (again, should be same for all surface fluxes). If `uₕ` consists of the covariant velocity components, it's not quite right (it should be the velocity vector consisting of the contravariant components--is that what it is?)

Reply:
No, Y.c.uₕ is the covariant velocity (horizontal). You are right this is not correct, I just didn't write the equation carefully. Our current implementation of surface flux should be correct. `ρatke_flux` is also not a WVector (vertical), instead it should be a Covariant3Vector, but with the physical magnitude as in the equation, right?

Reply:
> But the variables are not on the same spaces (some are on centers and some are on faces), so we may need to use the DataLayouts? @charleskawczynski @dennisYatunin any idea?

On the GPU we'll need to, because `bycolumn` is functionally different (https://github.com/CliMA/ClimaCore.jl/commit/b1bdf1e6db35b9ae3fb6a4fe358f27a3df4c840d). Maybe @simonbyrne can re-visit making `bycolumn` on the GPU functionally the same (and maintain performance). Doing that would be really nice because then we can just use bycolumn and use field broadcasting (PointFields can broadcast like scalars do, so they don't need to be on the same datalayout).